### ISSUE TITLE
Make Serializers external

### ISSUE THREAD

**Issue by wzorgdrager** [codefeedr/codefeedr#163]
Currently there is no option to add an external SerDe and types are defined through an Enum. This could be a useful addition.

**Comment by wzorgdrager** [comment]
Potential approach; add a registry to the `Serializer` class, if the name cannot be found then look in the buffer registry.

**Comment by wzorgdrager** [comment]
This is now addressed in #156 
Works like this:
```scala
//define your Serde
Serializer.register[MySerde[_]]("my_serde")

//do magic

pipelineBuilder.setBufferProperties(Buffer.SERIALIZER, "my_serde")
```
Needs to be properly documented though. Also, this is not the prettiest approach but for now it works.
